### PR TITLE
Generate transfers.txt

### PR DESCRIPTION
**Pull request by datanel**
```
cargo build
target/debug/transfe_rs -i path/to/stops.txt
```

This will generate a GTFS transfers.txt file in the root of the project
you can also specify the target
```
target/debug/transfe_rs -i path/to/stops.txt -o path/to/transfers.txt
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # transfe_rs
 
+[![Build Status](https://travis-ci.org/CanalTP/transfe_rs.svg?branch=master)](https://travis-ci.org/CanalTP/transfe_rs)
```
Comment (pbougue) [inline README.md]: to rebase, as it is already in readme

File: src/main.rs [hunk 2]
```diff
@@ -22,13 +25,67 @@ struct Args {
     flag_output: String,
 }
 
+#[derive(RustcDecodable)]
```
Comment (antoine-de) [inline src/main.rs]: Isn't rustc_serialize deprecated over serde ?

File: src/main.rs [hunk 3]
```diff
@@ -22,13 +25,67 @@ struct Args {
...
+struct StopPoint {
+    stop_id: String,
+    // stop_code: Option<String>,
+    _stop_name: String,
```
Comment (antoine-de) [inline src/main.rs]: why this `_` before the variable name ?

File: src/main.rs [hunk 4]
```diff
@@ -22,13 +25,67 @@ struct Args {
...
-             args.flag_output);
+        .and_then(|dopt| dopt.decode())
+        .unwrap_or_else(|e| e.exit());
+
+    let mut rdr = csv::Reader::from_file(args.flag_input)
+        .unwrap()
+        .double_quote(true);
+
+    let mut stop_point_list = vec![];
+    for record in rdr.decode() {
```
Comment (antoine-de) [inline src/main.rs]: I think a ggod exercice would be to rewrite this loop in a more rustic way with iterators and filter :stuck_out_tongue_winking_eye:

File: src/main.rs (hunk 3 above)
Comment (pbougue) [inline src/main.rs]: because it's optional in GTFS (and the point is to avoid reading them in the future)

Comment (pbougue) [inline src/main.rs]: so these fields are gonna disappear

Comment (datanel) [inline src/main.rs]: without _ when building there are warnings. variables no used. 
As we don't use these values we hide the warnings

File: README.md (hunk 1 above)
Comment (datanel) [inline README.md]: rebased

File: src/main.rs (hunk 2 above)
Comment (TeXitoi) [inline src/main.rs]: Not until serde works reliably on stable. In this case, that's because rust-csv use rustc_serialize.

File: src/main.rs (hunk 4 above)
Comment (TeXitoi) [inline src/main.rs]: In this case, look at `Iterator::filter`, `Iterator::map`, `Iterator::filter_map`, `Iterator::collect`.

Reading the doc of `Iterator`, `Option` and `Result`is very important.

File: src/main.rs [hunk 5]
```diff
@@ -22,13 +26,69 @@ struct Args {
...
+        .and_then(|dopt| dopt.decode())
+        .unwrap_or_else(|e| e.exit());
+
+    let mut rdr = csv::Reader::from_file(args.flag_input)
+        .unwrap()
+        .double_quote(true);
+
+    let stop_point_list: Vec<StopPoint> = rdr.decode()
+        .filter_map(|rc| {
+            rc.map_err(|e| {
```
Comment (datanel) [inline src/main.rs]: not sure this is the best way for handling errors 
@TeXitoi ?

Comment (TeXitoi) [inline src/main.rs]: Depends what you want. To just log buggy lines, that'll do the job. You can use https://crates.io/crates/log for better logging.

You can also panic (unwrap or expect).

You can also write a function that'll return a `Result<Vec<StopPoint>, csv::Error>` if you want fine grained error handling.

**Comment by antoine-de** [comment]
OK for me

**Pull request merged**